### PR TITLE
feature required actions

### PR DESCRIPTION
**Pull request by orteth01**

### CODE REVIEW

**Comment by orteth01** [comment]
sample output

> You have 97 total required actions. 29 conversations require an action. 3 conversations have a specific buyer request. 1 items require an action. 1 items Require Attribution Verification. 67 transactions require an action. 44 transactions can be marked as shipped. 5 transactions are buyer orders. 1 transactions can be marked as ready for pick up.

**Comment by orteth01** [comment]
i just have everything plural for now (e.g. 1 items). i can handle when there is just a singular item/conversation/transaction later, if you want

**Comment by brettjashford** [comment]
> sample output

dope!

> i just have everything plural for now (e.g. 1 items). i can handle when there is just a singular item/conversation/transaction later, if you want

yeah i was planning on doing the same

File: lib/getRequiredActions/utils.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+const TRANSACTION_ACTION_REASONS = {
+  // not 100% sure these things all mean this, but for demo purposes i think it's okay
```
Comment (brettjashford) [inline lib/getRequiredActions/utils.js:2]: i think that's correct, yeah should be good enough

File: lib/getRequiredActions/utils.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  // map to count the total number of items for each rejection reason
+  const rejectionCountMap = items.edges.reduce((rejectionCountMap, edge) => {
+    const item = edge.node;
+    const sortedReviews = item.reviews.documents
+      .filter(review => review.type === 'EXPERT_REVIEW')
+      .sort((a, b) => {
+        const aModifiedDate = a.modifiedDate;
+        const bModifiedDate = b.modifiedDate;
+        return dateStringToMilli(bModifiedDate) - dateStringToMilli(aModifiedDate);
```
Comment (brettjashford) [inline lib/getRequiredActions/utils.js:54]: this need to be defined somewhere?

Comment (orteth01) [reply lib/getRequiredActions/utils.js:54]: whoooops yes

Comment (orteth01) [reply lib/getRequiredActions/utils.js:54]: added

**Pull request merged**